executor/analyze.go

```diff
@@ -1066,11 +1070,17 @@ func (e *AnalyzeColumnsExec) buildSamplingStats(
 		oldRootCollectorSize := rootRowCollector.Base().MemSize
 		rootRowCollector.MergeCollector(mergeResult.collector)
 		e.memTracker.Consume(rootRowCollector.Base().MemSize - oldRootCollectorSize - mergeResult.collector.Base().MemSize)
+		ReleaseMemory(e.memTracker, oldRootCollectorSize+mergeResult.collector.Base().MemSize-rootRowCollector.Base().MemSize)
 	}
 	if err != nil {
 		return 0, nil, nil, nil, nil, err
 	}
 
+	logutil.BgLogger().Info("memory:", zap.Int64("rootCollector", rootRowCollector.Base().MemSize))
+	memStats := &runtime.MemStats{}
+	runtime.ReadMemStats(memStats)
```

Comment:
```runtime.ReadMemstats ``` will cause the ```Stop-The-World``` to affect the performance. We should use a global ReadMemstats to get mem stats and share all data to every component.